### PR TITLE
[COST-451] AWS Tag Enablement

### PR DESCRIPTION
**Pull request by cgoodfred**
This PR is targeted at allowing AWS tags that are enabled/disabled to reflect those changes in the daily summary. It also puts new tags into the enabledtagkeys table to make them enabled by default. 

The data migration was done to fill the `aws_enabledtagkeys` table in production since the sql checks to see if a tag exists in the `aws_tagsummary` table before adding it to the enabled tag keys table. It does this to prevent adding keys that already exist and have been disabled when a report file is processing with those keys in it.

Testing:

1. Generate some nise data, (here is an example yml but you will need to remove the .txt, github doesn't let you upload .yml's):
[aws_tags_enabled_testing.yml.txt](https://github.com/project-koku/koku/files/5440638/aws_tags_enabled_testing.yml.txt)
```nise report aws --static-report-file aws_tags_enabled_testing.yml --aws-s3-bucket-name testing/local_providers/aws_local_0 --aws-s3-report-name testingAwsTags```

2. create a source for the data
3. Go check out the database and check the `reporting_awsenabledtagkeys` table and make sure all the tags are there, there should be 5:
 ```
id |     key     
----+-------------
  1 | crazy
  2 | environment
  3 | show
  4 | title
  5 | trash
(5 rows)
```
4. Check out the tags on the `reporting_awscostentrylineitem_daily_summary` table and make sure you see the appropriate distinct tags for the data you use, here are the ones from the example yml:
```
                  tags                   
-----------------------------------------
 {"show": "house", "environment": "dev"}
 {"trash": "wednesday"}
 {"crazy": "dogs"}
 {"title": "green", "trash": "tuesday"}
(4 rows)
```

5. Note the total costs for aws:  `http://127.0.0.1:8000/api/cost-management/v1/reports/aws/costs/`

6. Disable (delete it from the `reporting_awsenabledtagkeys` table) some tags, I have removed the trash tag from my example

7. Resummarize the data

8. Check the distinct tags from the `reporting_awscostentrylineitem_daily_summary` table again and make sure the tags do not include the disabled tag. The example now looks like the following:
```
                  tags                   
-----------------------------------------
 {"title": "green"}
 
 {"show": "house", "environment": "dev"}
 {"crazy": "dogs"}
(4 rows)
```
9. Check the aws cost and ensure that they have not changed.

10. Also test OCP on AWS with a similar method.

### CODE REVIEW

File: koku/masu/database/sql/reporting_awscostentrylineitem_daily_summary.sql [hunk 1]
```diff
@@ -1,5 +1,27 @@
...
+        FROM (
+            SELECT lid.id,
+                lid.tags as aws_tags,
+                aak.key_array
+            FROM {{schema | sqlsafe}}.reporting_awscostentrylineitem_daily lid
+            JOIN cte_array_agg_keys aak
+                ON 1=1
+            WHERE (
+                    lid.tags ?| aak.key_array
+                )
```
Comment (adberglund) [inline koku/masu/database/sql/reporting_awscostentrylineitem_daily_summary.sql:19]: We'll want to add start date / end date filtering and bill id filtering here so we only grab from the the source we're interested in and the date range we're looking at.

File: koku/masu/database/sql/reporting_ocpawscostlineitem_daily_summary.sql [hunk 2]
```diff
@@ -94,6 +96,62 @@ CREATE TEMPORARY TABLE matched_tags_{{uuid | sqlsafe}} AS (
...
+        FROM (
+            SELECT lid.id,
+                lid.tags as aws_tags,
+                aak.key_array
+            FROM {{schema | sqlsafe}}.reporting_awscostentrylineitem_daily lid
+            JOIN cte_array_agg_keys aak
+                ON 1=1
+            WHERE (
+                    lid.tags ?| aak.key_array
+                )
```
Comment (adberglund) [inline koku/masu/database/sql/reporting_ocpawscostlineitem_daily_summary.sql:118]: We'll want to add start date / end date filtering and bill id filtering here so we only grab from the the source we're interested in and the date range we're looking at.

**Review by adberglund — CHANGES_REQUESTED** [review]
Just some additions to some SQL WHERE clauses.

File: koku/masu/database/sql/reporting_awscostentrylineitem_daily.sql [hunk 3]
```diff
@@ -121,5 +121,17 @@ INSERT INTO {{schema | sqlsafe}}.reporting_awscostentrylineitem_daily (
     FROM reporting_awscostentrylineitem_daily_{{uuid | sqlsafe}}
 ;
 
+INSERT INTO {{schema | sqlsafe}}.reporting_awsenabledtagkeys (
```
[the remark this answers is not in the document]
Comment (cgoodfred) [reply koku/masu/database/sql/reporting_awscostentrylineitem_daily.sql:124]: Changed the FROM to be `reporting_awscostentrylineitem_daily_{{uuid | sqlsafe}}`

File: koku/masu/database/sql/reporting_awscostentrylineitem_daily_summary.sql (hunk 1 above)
Comment (cgoodfred) [reply koku/masu/database/sql/reporting_awscostentrylineitem_daily_summary.sql]: added start/end/bill id filtering

File: koku/masu/database/sql/reporting_ocpawscostlineitem_daily_summary.sql (hunk 2 above)
Comment (cgoodfred) [reply koku/masu/database/sql/reporting_ocpawscostlineitem_daily_summary.sql]: added start/end/bill id filtering

**Review by adberglund — APPROVED** [review]
🎃

**Pull request merged**